src/Toml/Bi/Combinators.hs

```diff
@@ -298,3 +301,17 @@ list codec key = Codec
   where
     nonEmptyCodec :: TomlCodec (NonEmpty a)
     nonEmptyCodec = nonEmpty codec key
+
+-- | 'Codec' for set of values. Represented in TOML as array of tables.
+set :: forall a . Ord a => TomlCodec a -> Key -> TomlCodec (Set a)
+set codec key = dimap S.toList S.fromList listCodec
+            where
+              listCodec :: TomlCodec [a]
+              listCodec = list codec key
```

Comment:
I think that having a separate variable under `where` is not neccessary here, and you can write the implementation in a single line:

```haskell
set codec key = dimap S.toList S.fromList (list codec key)
```